capa/rules.py

```diff
@@ -276,27 +276,66 @@ def parse_description(s, value_type, description=None):
     return value, description
 
 
+def get_statement_description_entry(d):
+    """
+    extracts the description for statements
+    a statement must only have one description
+
+    example:
+    the features definition
+      - or:
+        - description: statement description
+        - number: 1
+          description: feature description
+
+    becomes
+      <statement>: [
+        { "description": "statement description" },  <-- extracted here
+        { "number": 1, "description": "feature description" }
+      ]
+    """
+    if not isinstance(d, list):
+        return None
+
+    # identify child of form '{ "description": <description> }'
+    descriptions = list(filter(lambda c: isinstance(c, dict) and len(c) == 1 and "description" in c, d))
+    if len(descriptions) > 1:
+        raise InvalidRule("statements can only have one description")
+
+    if not descriptions:
+        return None
+
+    return descriptions[0]
+
+
 def build_statements(d, scope):
     if len(d.keys()) > 2:
         raise InvalidRule("too many statements")
 
     key = list(d.keys())[0]
+
+    description = get_statement_description_entry(d[key])
```

Comment:
thoughts on doing: `description = pop_statement_description(d[key])` (and make this routine modify its parameter, removing any returned entry).

rational might be to keep logic around description handling within a single function, but its not a blocker.